From fa37e6e0713f4a548fe2bcf67fea3391abd51ad8 Mon Sep 17 00:00:00 2001
From: David Bauer <mail@david-bauer.net>
Date: Mon, 1 Jun 2026 00:08:55 +0200
Subject: [PATCH] ath79: force SGMII SerDes mode to MAC operation

The mode on the SGMII SerDes on the QCA9563 is 1000 Base-X by default.
This only allows for 1000 Mbit/s links, however when used with an SGMII
PHY in 100 Mbit/s link mode, the link remains dead.

This strictly has nothing to do with the SerDes calibration, however it
is done at the same point in the QCA reference U-Boot which is the
blueprint for everything happening here. As the current state is more or
less a hack, this should be fine.

This fixes the issues outlined above on a TP-Link EAP-225 Outdoor.

Reported-by: Tom Herbers <freifunk@tomherbers.de>
Tested-by: Tom Herbers <freifunk@tomherbers.de>
Submitted-by: David Bauer <mail@david-bauer.net>
---
 arch/mips/include/asm/mach-ath79/ar71xx_regs.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/include/asm/mach-ath79/ar71xx_regs.h b/arch/mips/include/asm/mach-ath79/ar71xx_regs.h
index 7fc359e2fac3b..0d2886a0a9fe8 100644
--- a/arch/mips/include/asm/mach-ath79/ar71xx_regs.h
+++ b/arch/mips/include/asm/mach-ath79/ar71xx_regs.h
@@ -1376,5 +1376,6 @@
 
 #define QCA956X_SGMII_CONFIG_MODE_CTRL_SHIFT	0
 #define QCA956X_SGMII_CONFIG_MODE_CTRL_MASK	0x7
+#define QCA956X_SGMII_CONFIG_MODE_CTRL_SGMII_MAC	0x2
 
 #endif /* __ASM_MACH_AR71XX_REGS_H */
